colossalai/kernel/cuda_native/flash_attention.py

```diff
@@ -86,11 +86,11 @@ def backward(ctx, grad_output):
 
     class ColoAttention(torch.nn.Module):
 
-        def __init__(self, embed_dim: int, num_heads: int, dropout: float = 0.0):
+        def __init__(self, embed_dim: int, num_heads: int, dropout: float = 0.0, scale=1.0):
             super().__init__()
             assert embed_dim % num_heads == 0, \
                 f"the embed dim ({embed_dim}) is not divisible by the number of attention heads ({num_heads})."
-            self.scale = 1 / math.sqrt(embed_dim // num_heads)
+            self.scale = scale
```

Comment:
How about check if the input scale is None, we use 1 / sqrt(embed_dim / num_heads), else we set its value

Reply:
> How about check if the input scale is None, we use `1 / sqrt(embed_dim / num_heads)`, otherwise we set its value

updated